internal/names/names.go

```diff
@@ -5,46 +5,48 @@ package names
 import (
 	"crypto/md5" // nolint:gosec // Non-crypto use
 	"encoding/hex"
+	"regexp"
 	"strings"
 
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
-// ReleaseName returns the name of a helm release derived from the
-// base string.
-func ReleaseName(base string) string {
-	return TruncateMD5(base, 53)
-}
-
-// ConfigurationName returns the name of a kube configuration derived from the
-// base string. It ensures that things like leading digits are
-// sufficiently hidden to prevent kube from erroring out on the name.
-func ConfigurationName(base string) string {
-	return GenerateResourceName("s-" + base)
-}
+var allowedDNSLabelChars = regexp.MustCompile("[^-a-z0-9]*")
 
-// IngressName returns the name of a kube ingress derived from the
-// base string. It ensures that things like leading digits are
-// sufficiently hidden to prevent kube from erroring out on the name.
-// It also replaces "/" characters with "-" to produce a valid resource
-// name from a route (which may contain "/" characters).
-func IngressName(base string) string {
-	baseSafe := strings.ReplaceAll(base, "/", "-")
-	return GenerateResourceName("i-" + baseSafe)
+// DNSLabelSafe filters invalid characters and returns a string that is safe to use as a DNS label.
+// It does not enforce the required string length, see `Sanitize`.
+//
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names
+func DNSLabelSafe(name string) string {
+	name = strings.Replace(name, "_", "-", -1)
+	name = strings.ToLower(name)
+	name = allowedDNSLabelChars.ReplaceAllLiteralString(name, "")
+	name = strings.TrimLeft(name, "-")
+	name = strings.TrimRight(name, "-")
+	return name
 }
 
 // GenerateResourceName joins the input strings with dots (".")  and
 // returns the result, suitably truncated to the maximum length of
 // kube resource names.
 func GenerateResourceName(names ...string) string {
-	return TruncateMD5(strings.Join(names, "."), 63)
+	name := DNSLabelSafe(strings.Join(names, "-"))
```

Comment:
As said here: https://github.com/epinio/helm-charts/pull/187/files#diff-7f0e1fad4fc65b5c6294727baa852f191201feafe9d27c9c0ff5dd28da4ba9beR28
replacing characters could result in collisions. I start thinking that we should simplify that down to just a  md5sum (truncated if needed, although it's always 32 characters long so should be small enough).

Reply:
Maybe always append an MD5, even if the incoming string is short enough ?
In that case the collision in the prefix does not matter because the md5 should be computed from the unmodified string, thus be different, and resolve the collision.

Reply:
As we agreed while pairing, we will always use the sha1sum of the original name and prefix it with as many characters are still free from the original name (with invalid characters removed).